Add docker structure

mesilov opened the pull request:
| Q             | A                                                                                                                         |
|---------------|---------------------------------------------------------------------------------------------------------------------------|
| Bug fix?      | no                                                                                                                    |
| New feature?  | yes <!-- please update CHANGELOG.md file -->                                                                           |
| Deprecations? | no <!-- please update CHANGELOG.md file -->                                                                           |
| Issues        | Fix #6 ... <!-- prefix each issue number with "Fix #", no need to create an issue if none exists, explain below instead --> |
| License       | MIT                                                                                                                       |

<!--
Replace this notice by a description of your feature/bugfix.
This will help reviewers and should be a good start for the documentation.

Additionally:
 - Always add tests and ensure they pass.
 - For new features, provide some code snippets to help understand usage.
-->

Commit:
Добавил контейнер с Postgres

KarlsonComplete committed:
- Добавил контейнер с nginx,php-fpm и postgres.

KarlsonComplete committed:
- Добавил расширения mbstring,zip,curl,bcmath

mesilov committed:
Merge pull request #12 from KarlsonComplete/add-docker-structure

Add docker containers

mesilov committed:
Add PHP CLI service and improve Docker setup

Added a PHP CLI service definition in `docker-compose.yaml` and created the appropriate Dockerfile. Enhanced Composer dependencies and updated Makefile with new commands for initializing and managing the application containers. Updated `.gitignore` to exclude Docker database files.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

KarlsonComplete committed:
- Добавил глобальные переменные

mesilov commented on docker/nginx/conf.d/nginx.conf:
```diff
@@ -0,0 +1,20 @@
+server {
```
А зачем нам nginx?

mesilov commented on docker/nginx/Dockerfile:
```diff
@@ -0,0 +1,5 @@
+FROM nginx:1.27-alpine
```
А зачем нам nginx?

mesilov commented on docker/php-fpm/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+FROM php:8.2-fpm-alpine
```
А зачем нам php-fpm?

mesilov commented on index.php:
```diff
@@ -0,0 +1,17 @@
+<?php
```
мы не работаем с веб-сервером, файл лишний

mesilov commented on composer.json:
```diff
@@ -36,14 +42,21 @@
+    "symfony/cache": "^7",
+    "symfony/console": "^7",
+    "symfony/dotenv": "^7",
+    "symfony/framework-bundle": "^7"
```
а зачем нам symfony/framework-bundle?
грохни и посмотри что упадёт

mesilov commented on docker/postgres/Dockerfile:
```diff
@@ -0,0 +1,12 @@
+FROM postgres:16.4-alpine
+
+# Устанавливаем переменные окружения для настройки базы данных
+ENV POSTGRES_USER=kirill
```
все явки-пароли через env-окружение

mesilov commented on docker-compose.yaml:
```diff
@@ -0,0 +1,49 @@
+    image: postgres:${POSTGRES_VERSION:-15}-alpine
+    restart: unless-stopped
+    environment:
+      POSTGRES_DB: ${POSTGRES_DB:-b24AccountTest}
```
все явки-пароли через env-окружение

mesilov commented:
@KarlsonComplete забирает актуальную версию веточки и открывает PR на неё из своей

mesilov committed:
Add UUID and member ID properties to Bitrix24Account

Updated the Bitrix24Account class with UUID and member ID properties, leveraging Symfony and Doctrine annotations. Additionally, replaced the deprecated SDK package and adjusted the PHPUnit testsuite configuration.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov commented on Makefile:
```diff
@@ -4,6 +4,10 @@
+#!/usr/bin/env make
+
+export COMPOSE_HTTP_TIMEOUT=120
+export DOCKER_CLIENT_TIMEOUT=120
```
Линтеры пускать через докер!

camaxtly replied:
Убрать тесты, которые запускаются не в докере

mesilov committed:
Refactor Docker setup and Bitrix24Account entity

Removed custom Docker configurations for nginx, PHP-FPM, and PostgreSQL to streamline the Docker environment. Introduced a centralized environment configuration using .env and Symfony Dotenv. Additionally, refactored the Bitrix24Account entity and its repository to enhance code maintainability and align with best practices.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update .gitignore to exclude /docker/postgres directory.

Added /docker/postgres directory to the .gitignore file to prevent it from being tracked by Git. This ensures that local PostgreSQL database files are not included in the repository.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add custom repository to Bitrix24Account

This change integrates the Bitrix24AccountRepository with the Bitrix24Account entity. The repository will handle database interactions specific to Bitrix24Account instances. This update enhances the modularity and maintainability of the codebase.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add functional test for Bitrix24AccountRepository

Introduced `Bitrix24AccountRepositoryTest` to ensure the repository's correct functionality. Modified the Makefile to run the functional tests suite with more detailed output including warnings.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove /docker/postgres from .gitignore

The /docker/postgres directory is no longer required to be ignored by Git. This change ensures that any important files in this directory can now be tracked and versioned.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor repository and entity manager usage.

Standardize usage of EntityManagerInterface and improve type inconsistencies. Apply `#[Override]` attribute in tests, and add null checks and validation for better robustness.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update PHPStan command in GitHub Actions workflow

Replaced the 'make lint-phpstan' command with a direct call to 'vendor/bin/phpstan --memory-limit=2G analyse.' This change ensures better control over memory usage and eliminates the dependency on the Makefile.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update Rector command in GitHub workflow

Replaced `make lint-rector` with `vendor/bin/rector process --dry-run` for better control and customization of the Rector tool in CI pipeline. This change ensures a direct invocation of Rector with specified options without intermediary make tasks.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Create functional test workflow and update configurations

Add a new GitHub workflow for functional tests with specified environments and steps. Update `docker-compose.yaml` to reflect the new PostgreSQL database settings. Rename existing workflows for clarity and consistency.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Switch to direct phpunit command for unit tests

Replaced the "make test-unit" command with a direct call to phpunit in the GitHub Actions workflow for better clarity and control over the test execution. This change uses the phpunit command with options for test suite, warnings, and formatted output.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update functional test workflow to include DB setup

Modified the functional test workflow to drop and recreate the database schema before running tests. This ensures a consistent test environment and improves test reliability.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update DB env variables in workflow configuration

Changed the environment variable names from POSTGRES_* to DATABASE_* in the tests-functional.yml file. This ensures consistency with other parts of the project and provides more descriptive variable names.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add database configuration to functional test workflow

Include essential environment variables for the PostgreSQL database setup within the functional test GitHub Actions workflow. This ensures the tests have access to the correct database credentials.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add new use cases for Bitrix24 account management

Implemented commands and handlers for InstallStart, InstallFinish, Uninstall, ChangeDomainUrl, and RenewAuthToken use cases. Added testing and improved event handling in Bitrix24Account class.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove redundant flush argument from save method

Eliminated the unnecessary flush argument in the save method across multiple handlers and the Bitrix24AccountRepository. This refactor ensures the flush operation is consistently handled within the repository, simplifying the contract and usage.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor repository methods to include PHPStan type hints

Added detailed PHPStan type hints with @phpstan-return annotations to various repository methods to improve static analysis and type safety. Refactored array filtering logic in the RenewAuthToken handler for clarity and efficiency

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor account handling logic in multiple use cases

Updated the condition checks and variable names for clarity and consistency across various use cases such as Install, Uninstall, ChangeDomainUrl, and RenewAuthToken. Minor changes include replacing array length checks with empty array comparisons and standardizing the event dispatching process. Also, added missing #[Override] annotations in certain methods where applicable.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

KarlsonComplete committed:
- Стянул изменения из основного репозитория

KarlsonComplete committed:
- Добавил мапинг на столбцы: isBitrix24UserAdmin,domain_url,updatedAt,applicationVersion.

KarlsonComplete committed:
- Добавил мапинг на столбец: accountStatus.
- Пытался еще добавить на authToken.

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,160 @@
+    public function save(Bitrix24AccountInterface $bitrix24Account): void
+    {
+        $this->getEntityManager()->persist($bitrix24Account);
+        //todo discuss add flush arg to contract or add flusher in usecases?
```
Добавляем отдельный flusher отдельным сервисом Елисеев style. https://github.com/deworkerpro/demo-auction/blob/master/api/src/Flusher.php

camaxtly replied:
вынесли

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,160 @@
+    /**
+     * @param non-empty-string $applicationToken
+     * @phpstan-return array<Bitrix24AccountInterface&AggregateRootEventsEmitterInterface>
+     * @todo discuss are we need add this method in contract in b24phpsdk?
```
Базовый сценарий в sdk, надо вынести в контракты и нам здесь поддержать

mesilov replied:
https://github.com/bitrix24/b24phpsdk/issues/62

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Handler.php:
```diff
@@ -0,0 +1,72 @@
+            'b24_user_id' => $command->bitrix24UserId
+        ]);
+
+        //todo discuss are we need add bitrix24UserId in contract?
```
Имхо,  это редкая операция "закончить установку" и поэтому пока можно оставить как есть. Если на практике будут проблемы, пересмотреть решение

camaxtly commented on src/Bitrix24Accounts/UseCase/RenewAuthToken/Handler.php:
```diff
@@ -0,0 +1,85 @@
+        ]);
+
+        // get all active bitrix24 accounts
+        //todo discuss add bitrix24_user_id in contract?
```
Поднимаем контракт в sdk, можно сделать nullable по дефолту

mesilov replied:
https://github.com/bitrix24/b24phpsdk/issues/63

mesilov replied:
https://github.com/bitrix24/b24phpsdk/pull/69 исправлено

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,160 @@
+
+    public function findAllActive(): array
+    {
+        return $this->getEntityManager()->getRepository(Bitrix24Account::class)->findBy(
```
Добавить инструмент отвечающий за код-стайл

mesilov replied:
добавлен php-cs-fixer

mesilov committed:
Merge pull request #14 from KarlsonComplete/add-docker-structure

Add docker structure

KarlsonComplete committed:
- Добавил Flusher.php
- Добавил вызов flusher в Bitrix24AccountRepository.php , но что то получается не так.
- В функциональных тестах получается больше ошибок из за вызова flusher.

KarlsonComplete committed:
- Добавил в тестах при создании экземпляра Bitrix24AccountRepository второй параметр Flusher.

KarlsonComplete committed:
- Добавил отображения структуры бд (где это отображается?)

KarlsonComplete committed:
- Создал xml файл для маппинга.
- В EntityManagerFactory.php поменял на создание через xml
- В Makefile добавил отдельное создание и удаление бд.
- Для тестирования добавил тестовую сущность

mesilov committed:
Merge pull request #15 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly commented on composer.json:
```diff
@@ -62,7 +75,7 @@
   "autoload-dev": {
     "psr-4": {
       "Bitrix24\\SDK\\Lib\\Tests\\": "tests",
```
Убрать SDK из пространства имен.  У нас просто Bitrix24\Lib.  Соответственно проверить во всех конфигурационных файлах

camaxtly commented on src/Bitrix24Accounts/UseCase/Command/Flusher.php:
```diff
@@ -0,0 +1,17 @@
+
+use Doctrine\ORM\EntityManagerInterface;
+
+final readonly class Flusher
```
Flusher это сервис, он  должен быть "выше" уровнем чем UseCase, на одном уровне с Entity можно сделать папку Services и там уже размещать все сервисы

camaxtly commented on src/Bitrix24Accounts/ReadModel/Fetcher.php:
```diff
@@ -0,0 +1,39 @@
+use Knp\Component\Pager\Pagination\PaginationInterface;
+use Knp\Component\Pager\PaginatorInterface;
+
+class Fetcher
```
Фетчеров скорее всего будет много, поэтому тут тоже надо подумать как и на какие неймспейсы это все разложить

KarlsonComplete replied:
Переименовал фетчер в Bitrix24AccountFetcher, как я понимаю для каждой сущности будет свой фетчер

mesilov commented on src/Bitrix24Accounts/UseCase/RenewAuthToken/Handler.php:
```diff
@@ -0,0 +1,85 @@
+        );
+
+        if ($command->bitrix24UserId === null && count($accounts) > 1) {
+            //todo discuss move to b24phpsdk contracts?
```
https://github.com/bitrix24/b24phpsdk/issues/64

mesilov commented on src/Bitrix24Accounts/UseCase/Uninstall/Handler.php:
```diff
@@ -0,0 +1,51 @@
+        $this->logger->debug('Bitrix24Accounts.Uninstall.start', [
+            'b24_application_token' => $command->applicationToken,
+        ]);
+        //todo remove after update contract in b24phpsdk
```
https://github.com/bitrix24/b24phpsdk/issues/62

KarlsonComplete committed:
- Убрал из namespace sdk

KarlsonComplete committed:
- Переименовал xml файлы для маппинга
- Поменял подключение схем в xml
- Начал менять поля в xml

KarlsonComplete committed:
- Перенес Flusher
- Добавил useCase на сохранение
- Добавил xml с authtoken

KarlsonComplete committed:
- Добавил маппинг AuthToken для сущности Bitrix24Account

mesilov committed:
Merge pull request #16 from KarlsonComplete/add-docker-structure

Add docker structure

mesilov commented on composer.json:
```diff
@@ -28,26 +33,34 @@
-    "doctrine/orm": "^2",
-    "doctrine/doctrine-bundle": "^2",
-    "doctrine/doctrine-migrations-bundle": "^3",
+    "bitrix24/b24phpsdk": "^1",
```
переключиться на dev-dev, т.к. я исправления в SDK я волью в релиз 1.2.0, а он собирается dev-ветке

mesilov replied:
Исправлено

mesilov committed:
Update bitrix24/b24phpsdk dependency version

Changed the bitrix24/b24phpsdk dependency from version ^1 to dev-dev in composer.json. This ensures compatibility with the latest development version of the package, possibly addressing some bugs or adding new features.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add PHP-CS-Fixer configuration and update dependencies

Introduce a new PHP-CS-Fixer configuration file to ensure consistent code formatting. Update "friendsofphp/php-cs-fixer" to version 3.64 in composer.json. Also, add "*.cache" to .gitignore for better cache file management.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add GitHub Actions for PHP CS-Fixer

Introduce a GitHub Actions workflow to run PHP CS-Fixer for code linting on push and pull request events. Additionally, update the Makefile to include new targets for running and fixing code style issues with PHP CS-Fixer.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Enhance PHP-CS-Fixer configuration with additional rules

Updated .php-cs-fixer.dist.php to include cache file and new rule sets (@Symfony, @DoctrineAnnotation, @PhpCsFixer). Also adjusted the string concatenation formatting for better readability.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

KarlsonComplete committed:
- Добавил маппинг Scope для сущности Bitrix24Account

KarlsonComplete committed:
- Поменял название id на uuid в сущности и в маппинге

mesilov committed:
Merge pull request #17 from KarlsonComplete/add-docker-structure

Add docker structure

KarlsonComplete committed:
- Получается добавлять в таблицу

mesilov committed:
Merge pull request #18 from KarlsonComplete/add-docker-structure

Добавил маппинг , исправляю функциональные тесты

mesilov committed:
Add environment variable loading and debugging support

This commit updates the Makefile to include default and personal environment variable files. It introduces a new debug-print-env target to print essential environment variables for debugging purposes. Docker compose file adjustments and minor code formatting tweaks are also included.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add functional test for ChangeDomainUrl handler

Introduce functional testing for the ChangeDomainUrl handler to ensure the domain URL is updated correctly. Included flush operation in the handler to persist changes.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add flusher support to InstallFinish handler

Introduced a Flusher service to the InstallFinish handler to ensure database changes are persisted immediately. Updated the HandlerTest to include a functional test for the new behavior, ensuring that application tokens are validated correctly after persistence.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add functional test for InstallStart handler

Introduced a new PHPUnit test to ensure the correctness of the `InstallStart` handler's functionality. Also, integrated the `Flusher` service within the handler to persist changes.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove custom exception and update handler logic

Removed `MultipleBitrix24AccountsFoundException` and adjusted handler logic to use the existing exception from the SDK. This ensures consistency in exception handling across the application and adds the `Flusher` service to the token renewal handler for persisting changes.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove SaveAccount use case and add proper Uninstall testing

Deleted the SaveAccount use case and related tests to streamline the project. Enhanced the Uninstall use case by adding a unit test for the handler and integrating the Flusher service for better transaction management.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add TraceableEventDispatcher to HandlerTests

Refactor HandlerTests to use TraceableEventDispatcher, enabling better tracking of events. This change improves the readability and maintainability of test cases by using a consistent event dispatching mechanism.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update test to check for domain URL change event

Refactor HandlerTest to assert that the Bitrix24AccountDomainUrlChangedEvent is triggered instead of Bitrix24AccountApplicationInstalledEvent. This ensures the correct event is dispatched when the domain URL changes.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove unused import in HandlerTest.php

Removed the unused `Bitrix24AccountApplicationInstalledEvent` import in the `ChangeDomainUrl/HandlerTest.php` file. This cleanup helps reduce code clutter and improve readability.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

KarlsonComplete committed:
- Убрал автогенерацию
- Пофиксил статус при тестах

KarlsonComplete committed:
- Поменял тип в маппинге с datetime_immutable на carbon_immutable

mesilov committed:
Merge pull request #19 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -240,8 +293,8 @@ public function getUpdatedAt(): CarbonImmutable
     {
-        if (Bitrix24AccountStatus::active !== $this->accountStatus) {
-            throw new InvalidArgumentException(sprintf('account must be in status «active», but now account in status «%s»', $this->accountStatus->name));
+        if (Bitrix24AccountStatus::active !== $this->status) {
```
Можно ли сделать хелперы для enum?

camaxtly replied:
@camaxtly нид чек/выработка решения

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -282,18 +339,33 @@ public function markAsActive(?string $comment): void
+        $this->status = Bitrix24AccountStatus::blocked;
         $this->comment = $comment;
         $this->updatedAt = new CarbonImmutable();
+        $this->events[] = new Bitrix24AccountBlockedEvent(
```
Коммент пробрасываем с событие блокировки

mesilov replied:
https://github.com/bitrix24/b24phpsdk/pull/80

added

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,165 @@
+            $criteria['isBitrix24UserAdmin'] = $isAdmin;
+        }
+
+        return $this->findBy($criteria);
```
Проверить тип данных возвращаемого результата, когда 1 или несколько элементов найдено

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,165 @@
+     * @inheritdoc
+     */
+    #[Override]
+    public function delete(Uuid $uuid, bool $flush = false): void
```
Вынести Flusher из репозитория

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,165 @@
+        }
+    }
+
+    public function findAllActive(): array
```
Или оборачиваем в генератор, либо лимит/офсет

KarlsonComplete replied:
Добавил в функцию две переменных limit и offset

camaxtly commented on src/Bitrix24Accounts/UseCase/ChangeDomainUrl/Handler.php:
```diff
@@ -0,0 +1,46 @@
+            $account->changeDomainUrl($command->newDomainUrlHost);
+            $this->bitrix24AccountRepository->save($account);
+            $this->flusher->flush();
+            // todo выяснить почему он не видит объединение типов
```
Ну так и разберитесь, что там =)))

KarlsonComplete replied:
@mesilov , @camaxtly не особо понимаю в чем тут проблема.

KarlsonComplete committed:
- Убрал в аккаунте атрибуты
- Добавил в тесте обновления токена вызов flusher
- Убрал из репозитория вызовы flusher

mesilov committed:
Merge pull request #20 from KarlsonComplete/add-docker-structure

- Убрал в аккаунте атрибуты

mesilov committed:
Add FlusherDecorator for functional tests

Introduce the `FlusherDecorator` to facilitate flushing in functional tests. Updated `Bitrix24AccountRepositoryTest` to use this new decorator. Enhanced the Makefile to include an ORM schema update command for functional test execution.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor and enforce input validation in Bitrix24AccountRepository

Remove unused use statement and improve code formatting for consistency. Add input validation for non-empty strings and introduce error handling for invalid account statuses in delete method.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Fix formatting and simplify tests

Adjusted the whitespace formatting in Handler.php and Bitrix24Account.php for improved readability. Refactored HandlerTest.php to use Bitrix24AccountBuilder for cleaner test setup and added a new test case for handling domain URL changes across multiple accounts.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

KarlsonComplete committed:
- Добавил xdebug
- Изменил подход к EntityManager c фабрики на "синглтон"
- Добавил в makefile запуск одиночного теста с запуском xdebug

KarlsonComplete committed:
Merge pull request #21 from KarlsonComplete/add-docker-structure

Поправил некоторые ошибки в функциональных тестах

mesilov committed:
Refactor repository methods and improve exception handling

Formatted the Bitrix24AccountRepository methods for readability and added null checks for better error handling. Introduced additional validation for application tokens and domain URLs to prevent empty string errors.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor exception formatting for readability

Reformat exception messages to use multi-line structure for better readability. This change affects the methods: applicationInstalled, uninstallApplication, and updateApplicationVersion. Additionally, updated Bitrix24AccountBlockedEvent instantiation to include a comment attribute.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor event dispatching mechanism

Revised the event dispatching mechanism to handle emitted events directly from Bitrix24Account entity. Also streamlined constructor formatting and updated doc annotations for consistency.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor `Handler.php` files for code clarity and maintainability

Adjusted import statements, formatted constructor and exception handling, and added PHPDoc comments for improved readability and maintenance. These changes enhance code consistency and readability across the `Uninstall`, `InstallStart`, and `InstallFinish` handlers.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor event dispatching mechanism

Revised the event dispatching mechanism to handle emitted events directly from the Bitrix24Account entity. Streamlined constructor formatting and updated doc annotations for consistency.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

KarlsonComplete commented on src/Bitrix24Accounts/UseCase/RenewAuthToken/Handler.php:
```diff
@@ -0,0 +1,77 @@
+        ]);
+
+        // get all active bitrix24 accounts
+        $accounts = $this->bitrix24AccountRepository->findByMemberId(
```
Они похожи , но все таки проверки различаются в зависимости от задач usecase, мне кажется лучше это не трогать

camaxtly replied:
Где-то надо зафиксировать, сотношение аккаунтов и приложения – много акков (на 1 приложение). Отсюда скорее всего и будет правильное понимание

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallStart/Handler.php:
```diff
@@ -0,0 +1,55 @@
+        $this->bitrix24AccountRepository->save($newAccount);
+        $this->flusher->flush();
+
+        foreach ($newAccount->emitEvents() as $event) {
```
@camaxtly посмотреть как можно не дублировать в каждом useCase

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -37,72 +44,92 @@ class Bitrix24Account implements Bitrix24AccountInterface
+        $this->expires = $authToken->expires;
         $this->applicationScope = $applicationScope->getScopeCodes();
+
+        if ($isEmitBitrix24AccountCreatedEvent) {
```
А можем ли обойтись без if-а

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -37,72 +44,92 @@ class Bitrix24Account implements Bitrix24AccountInterface
     public function getAuthToken(): AuthToken
     {
-        return new AuthToken($this->accessToken, $this->refreshToken, $this->expires);
+        $this->authToken = new AuthToken($this->accessToken, $this->refreshToken, $this->expires);
```
Если мы выделаем токен отдельным объектом, то и работать надо с ним как объектом

camaxtly commented on src/Bitrix24Accounts/ReadModel/Fetcher.php:
```diff
@@ -0,0 +1,39 @@
+
+class Fetcher
+{
+    public function __construct(
```
@KarlsonComplete нужны тесты на fetcher-ы, так как если переименуется стоолбец, то фетчер уже не будет работать.

KarlsonComplete replied:
добавил тест на проверку столбцов

camaxtly commented on tests/EntityManagerFactory.php:
```diff
@@ -0,0 +1,88 @@
+                Type::addType('carbon_immutable', CarbonImmutableType::class);
+            }
+
+            // $configuration = ORMSetup::createAttributeMetadataConfiguration($paths, $isDevMode);
```
отладку не стоит коммитить, не понятно, что тут нужно, а что нет

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/ChangeDomainUrl/HandlerTest.php:
```diff
@@ -0,0 +1,138 @@
+        $updated = $this->repository->getById($bitrix24Account->getId());
+        $this->assertEquals($newDomainUrl, $updated->getDomainUrl());
+
+        $this->assertTrue(in_array(
```
Для всех тестов добавляем комментарии, чтоб было видно, что пошло не так.

KarlsonComplete replied:
Добавил

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/RenewAuthToken/HandlerTest.php:
```diff
@@ -0,0 +1,96 @@
+        $this->assertEquals($newAuthToken->refreshToken, $updated->getAuthToken()->refreshToken);
+
+        // на продлении токенов событий не бросаем
+        $this->assertCount(
```
Кажется можно удалить проверку.

KarlsonComplete replied:
Удалил проверку

KarlsonComplete committed:
- Поменял подход к получению authToken
- Убрал комментарии
- Убрал одну из проверок при обновлении токена
- Удалил тестовую сущность User
- В cli-config.php заменил хардкод на переменную окружения

KarlsonComplete committed:
- Добавил сообщения для проверок в тестах

KarlsonComplete commented on src/Bitrix24Accounts/Entity/TestUser.php:
```diff
@@ -0,0 +1,44 @@
+
+namespace Bitrix24\Lib\Bitrix24Accounts\Entity;
+
+class TestUser
```
Убрал хвосты

KarlsonComplete committed:
- Добавил Fetcher тест
- В каждом use case сделал метод с проверкой
- Поправил в xml название столбца обновления

KarlsonComplete commented on src/Bitrix24Accounts/UseCase/InstallFinish/Handler.php:
```diff
@@ -0,0 +1,75 @@
+            'b24_user_id' => $command->bitrix24UserId
+        ]);
+
+        $accounts = $this->bitrix24AccountRepository->findByMemberId(
```
Добавил в классе handler отдельный метод чтобы там делать проверки. Изначально хотел добавить метод в репозиторий , так как у нас в других usecase где то проверки похожи , но все таки посчитал не очень хорошей идеей так как неудобно определять какой именно usecase сейчас работает.  То есть в исключениях не будет понятно что щас происходит обновление токена или завершение установки.

mesilov committed:
Merge pull request #22 from KarlsonComplete/add-docker-structure

Добавил fetcher тест и снял еще коменты

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -237,63 +269,103 @@ public function getUpdatedAt(): CarbonImmutable
+     * @return Event[]
+     */
+    #[\Override]
+    public function emitEvents(): array
```
унести в базовый класс, кидаем в корень src  – AggregateRoot и имплементирует AggregateRootEventsEmitterInterface

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,189 @@
+            );
+        }
+
+        $this->getEntityManager()->remove($bitrix24Account);
```
Делаем soft delete

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Handler.php:
```diff
@@ -0,0 +1,89 @@
+
+        $this->bitrix24AccountRepository->save($targetAccount);
+        $this->flusher->flush();
+        foreach ($targetAccount->emitEvents() as $event) {
```
@camaxtly покажи пример

KarlsonComplete committed:
- Война с AggregateRoot

KarlsonComplete committed:
- Внес правки для AggregateRoot
- Убрал из метода delete репозитория аккаунта хард делит и переделал на софт делит

KarlsonComplete committed:
- Добавил исключение в метод getById
- Поправил тесты

KarlsonComplete committed:
- Тест на фетчер

mesilov committed:
Merge pull request #23 from KarlsonComplete/add-docker-structure

Правки

camaxtly commented on src/Services/Flusher.php:
```diff
@@ -0,0 +1,17 @@
+
+use Doctrine\ORM\EntityManagerInterface;
+
+final readonly class Flusher
```
https://github.com/ElisDN/demo-project-manager/blob/master/manager/src/Model/Flusher.php

Можно сделать вот так

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -10,99 +10,112 @@
-    private int $expires;
-
-    private array $applicationScope;
+    private array|Scope $applicationScope;
```
Тут не понятно зачем? Имхо или одно, или другое

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -10,99 +10,112 @@
+        bool $isEmitBitrix24AccountCreatedEvent = false
+    ) {
+        $this->authToken = $authToken;
+        $array =  $applicationScope->getScopeCodes();
```
тут убрать после разбора

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -10,99 +10,112 @@
     }
 
-    #[Override]
+    public function setStatus(Bitrix24AccountStatus $status): void
```
метод вреден, так как меняет стейт в обход методов предметной области

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -117,87 +130,102 @@ public function renewAuthToken(RenewedAuthToken $renewedAuthToken): void
-     */
-    #[Override]
+    #[\Override]
     public function getApplicationScope(): Scope
```
Работаем с объектом

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -237,63 +269,96 @@ public function getUpdatedAt(): CarbonImmutable
-}
\ No newline at end of file
+
+    public function getEvents(): array
```
Должны были уйти в aggregateRoot

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -237,63 +269,96 @@ public function getUpdatedAt(): CarbonImmutable
+        return $this->events;
+    }
+
+    private function addAccountCreatedEventIfNeeded(bool $isEmitCreatedEvent): void
```
@camaxtly не забыть отдельно посмотреть зачем мы это делаем/какую проблему решаем

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,190 @@
+                )
+            );
+        }
+        $bitrix24Account->setStatus(Bitrix24AccountStatus::deleted);
```
Логика должна уехать внутрь класса аккаунта. А тут вызывать соответствующий метод

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,190 @@
+    }
+
+    #[Override]
+    public function delete(Uuid $uuid): void
```
@camaxtly а точно ли нам нужен?  Вернуться к вопросу после реализации useCase с удалением

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -0,0 +1,190 @@
+    }
+
+    #[Override]
+    public function save(Bitrix24AccountInterface $bitrix24Account): void
```
Пока подвесим вопрос с неймингом, вернемся после юзкейсов

camaxtly commented on src/Bitrix24Accounts/ReadModel/Fetcher.php:
```diff
@@ -0,0 +1,45 @@
+       // var_dump($queryBuilder->getMaxResults());
+        $query =  $this->em->createQuery("SELECT b24account FROM Bitrix24\Lib\Bitrix24Accounts\Entity\Bitrix24Account b24account");
+
+        return $this->paginator->paginate($query, $page, $size);
```
https://github.com/ElisDN/demo-project-manager/blob/master/manager/src/ReadModel/Work/Projects/Task/TaskFetcher.php

camaxtly commented on src/Bitrix24Accounts/UseCase/Uninstall/Handler.php:
```diff
@@ -0,0 +1,53 @@
+            $this->bitrix24AccountRepository->save($account);
+            $this->flusher->flush();
+
+            foreach ($account->emitEvents() as $event) {
```
Код должен уйти в Flusher

camaxtly commented on src/Bitrix24Accounts/UseCase/Uninstall/Handler.php:
```diff
@@ -0,0 +1,53 @@
+            $account->applicationUninstalled($command->applicationToken);
+            $this->bitrix24AccountRepository->save($account);
+            $this->flusher->flush();
+
```
Не хватает логов, что за акк и что произошло

camaxtly commented on src/Bitrix24Accounts/UseCase/Uninstall/Handler.php:
```diff
@@ -0,0 +1,53 @@
+            }
+        }
+
+        $this->logger->debug('Bitrix24Accounts.Uninstall.Finish');
```
Расширить контекст – число, ид юзера/портала

camaxtly commented on src/Bitrix24Accounts/UseCase/Uninstall/Handler.php:
```diff
@@ -0,0 +1,53 @@
+     */
+    public function handle(Command $command): void
+    {
+        $this->logger->debug('Bitrix24Accounts.Uninstall.start', [
```
Уровень infо

KarlsonComplete replied:
Поменял во всех хендлерах на info

camaxtly commented on src/Bitrix24Accounts/UseCase/Uninstall/Handler.php:
```diff
@@ -0,0 +1,53 @@
+        foreach ($accounts as $account) {
+            $account->applicationUninstalled($command->applicationToken);
+            $this->bitrix24AccountRepository->save($account);
+            $this->flusher->flush();
```
Зачем Flush в цикле?

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/Uninstall/HandlerTest.php:
```diff
@@ -0,0 +1,108 @@
+        $this->handler->handle(new Bitrix24Accounts\UseCase\Uninstall\Command($applicationToken));
+
+        $this->expectException(Bitrix24AccountNotFoundException::class);
+        $updated = $this->repository->getById($bitrix24Account->getId());
```
нейминг

KarlsonComplete committed:
- Правки на scope

mesilov committed:
Merge pull request #24 from KarlsonComplete/add-docker-structure

- Правки на scope

mesilov committed:
Simplify application scope handling

Updated the Bitrix24Account entity to store the application scope object directly instead of extracting scope codes. This change enhances flexibility for scope management and aligns with object-oriented design principles.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update PHP SDK in test commands

This commit adds a step to update the bitrix24/b24-php-sdk package using composer in the unit and functional test commands in the Makefile. This ensures the tests run with the latest version of the SDK, improving compatibility and catching any integration issues early.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update Composer flags to prefer source in workflows

Switch the Composer installation method from "prefer-dist" to "prefer-source" in unit and functional test workflows. This change aims to ensure that the source code is retrieved directly, which may be necessary for tracking changes or resolving specific dependency issues during testing.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add #[Override] attribute and refactor variable names

Added the #[Override] attribute to several function declarations to explicitly denote methods that override parent class methods. Refactored variable names across the project for improved clarity and consistency, particularly in test files and the Bitrix24Account entity. These changes enhance code readability and maintainability while aligning with coding standards.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Fix variable name in RenewAuthToken handler

Correct the variable name comment to reflect the accurate variable used in the code. This change clarifies the code by ensuring the comment and actual variable name are consistent, enhancing maintainability and readability.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update Composer flags and add specific package update

Change COMPOSER_FLAGS to use "prefer-dist" for general updates while ensuring bitrix24/b24-php-sdk uses "prefer-source" for more detailed control. This adjustment aims to optimize dependency handling and improve build efficiency.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Reorder composer update commands in CI workflow

Move the general composer update command to follow the specific package update in the CI configuration. This ensures that the more specific update of the bitrix24/b24-php-sdk package occurs first for clarity and task separation.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Reorder composer update commands.

Switched the order of composer update commands to ensure that environment-based flags are processed before specific package updates. This change maintains the configuration's flexibility while ensuring dependencies are correctly managed.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Reorder composer update commands.

Switched the order of composer update commands to ensure that environment-based flags are processed before specific package updates. This change maintains the configuration's flexibility while ensuring dependencies are correctly managed.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add step to install b24-php-sdk from source

This commit introduces an additional step in the GitHub Actions workflow to ensure the bitrix24/b24-php-sdk package is installed directly from source. This change is executed with options to prefer source installation and disable cache usage, which aids in maintaining the latest version of the package when running unit tests.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update SDK installation method in GitHub Actions workflow

Revised the method for installing the b24-php-sdk in the unit tests GitHub Actions workflow from 'composer update' to 'composer remove' followed by 'composer require'. This ensures a cleaner and more reliable installation from source, eliminating any potential issues with cached dependencies.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update SDK installation method in GitHub Actions workflow

Revised the method for installing the b24-php-sdk in the unit tests GitHub Actions workflow from 'composer update' to 'composer remove' followed by 'composer require'. This ensures a cleaner and more reliable installation from source, eliminating any potential issues with cached dependencies.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Update composer flags and install b24-php-sdk from source

Changed COMPOSER_FLAGS to prefer-dist to optimize dependency installation. Added steps to remove and require b24-php-sdk with prefer-source to ensure the latest source code is used. This configuration enhances the reliability of the workflow during functional tests.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Fix b24phpsdk installation naming and add to lint workflow

Corrected the naming of "b24phpsdk" installation steps in unit and functional test workflows. Added missing installation steps for "b24phpsdk" in the lint-phpstan workflow to ensure consistent dependency management across all workflows.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Refactor and streamline codebase

Removed unnecessary imports and redundant code to streamline multiple files. Improved compliance with coding standards by making syntax corrections and enforcing consistent code formatting. Additionally, added a Composer license checker configuration to manage dependencies.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove redundant b24phpsdk source installation steps.

The steps for installing b24phpsdk from source were unnecessary and have been removed from the unit tests, functional tests, and PHPStan lint workflows. This simplification should reduce maintenance overhead and speed up the workflow execution times.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove prefer-dist flag from Composer command

The prefer-dist flag has been removed from the COMPOSER_FLAGS in the unit test workflow configuration. This change allows for potentially more up-to-date installations by not limiting downloads to distribution packages. This can help ensure that the latest code is being tested.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove prefer-dist from COMPOSER_FLAGS in CI workflow.

Eliminating the --prefer-dist option can reduce reliance on distribution archives, allowing for more consistent test execution. This change aims to enhance the flexibility and potential versions available during the dependency resolution in the CI pipeline.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove xdebug from PHPUnit command in Makefile

The xdebug extension is no longer started with the PHPUnit command used for running functional tests. This change reduces overhead and improves performance during test execution.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Add GitHub workflow for license checking

This commit introduces a new GitHub Action workflow file named `license-check.yml` that checks for allowed licenses during code pushes and pull requests. It includes steps to set up PHP environments with specified versions and extensions, install highest or lowest dependencies, and run a license verification script. The workflow matrix supports running the checks on multiple PHP versions, ensuring license compliance across different environments.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

mesilov committed:
Remove PHP 8.2 from license check workflow

The license check workflow configuration no longer includes PHP 8.2 in the version matrix. This change focuses testing on PHP 8.3 and aims to streamline the maintenance of the workflow.

Signed-off-by: mesilov <mesilov.maxim@gmail.com>

KarlsonComplete committed:
- Починил Fetcher тест

KarlsonComplete committed:
- Переписал flusher и переписал
несколько тестов

mesilov committed:
Merge pull request #25 from KarlsonComplete/add-docker-structure

Update fetcher and flusher

camaxtly commented on .github/workflows/tests-functional.yml:
```diff
@@ -0,0 +1,84 @@
+          --health-timeout=5s
+          --health-retries=5
+        env:
+          POSTGRES_USER: b24phpLibTest
```
а нет ли здесь дублей?

KarlsonComplete replied:
убрал

camaxtly commented on src/Bitrix24Accounts/UseCase/ChangeDomainUrl/Command.php:
```diff
@@ -0,0 +1,19 @@
+
+readonly class Command
+{
+    public function __construct(
```
Где-то добавить проверку, чтоб не  поймать ошибку в рантайме.
Меняется код, а значит должен быть тест

KarlsonComplete replied:
Добавил

camaxtly commented on src/Services/Flusher.php:
```diff
@@ -0,0 +1,32 @@
+        foreach ($roots as $root) {
+            $events = $root->emitEvents();
+            foreach ($events as $event) {
+                var_dump($event);
```
отладка закралась

KarlsonComplete replied:
убрал

camaxtly commented on src/Services/Flusher.php:
```diff
@@ -0,0 +1,32 @@
+{
+    private $em;
+    private $eventDispatcher;
+    public function __construct(EntityManagerInterface $em,EventDispatcher $eventDispatcher) {
```
Контракт на интерфейс джля диспетчера

KarlsonComplete replied:
Добавил контракт симфони eventDispatcher или лучше сделать свой ?

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -0,0 +1,15 @@
+
+namespace Bitrix24\Lib\Bitrix24Accounts\UseCase\InstallFinish;
+
+readonly class Command
```
Нужны проверки на корректность команды, а что если пустые строки?

KarlsonComplete replied:
Сделал функцию validate и добавил тест , если все норм , то применю практику для всех остальных Command

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Handler.php:
```diff
@@ -0,0 +1,87 @@
+        ]);
+
+        /**
+         * @var AggregateRootEventsEmitterInterface|Bitrix24AccountInterface $bitrix24Account
```
Bitrix24AccountInterface – по идее в юзкейсе тут должен быть только этот

KarlsonComplete replied:
убрал лишнее

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Handler.php:
```diff
@@ -0,0 +1,87 @@
+
+        $this->bitrix24AccountRepository->save($bitrix24Account);
+        $this->flusher->flush();
+        foreach ($bitrix24Account->emitEvents() as $event) {
```
Удалить, оно уехало в Flusher

KarlsonComplete replied:
убрал

camaxtly commented on tests/Functional/Bitrix24Accounts/FetcherTest.php:
```diff
@@ -0,0 +1,70 @@
+        $this->repository = new Bitrix24AccountRepository($this->entityManager);
+    }
+
+    public function testListReturnsPaginatedResults(): void
```
Тут надо  тестировать то, что в нужные поля массива мы прочитали нужные нам строки

KarlsonComplete replied:
добавил еще один тест

KarlsonComplete committed:
- Поправил тест на изменения url домена

KarlsonComplete committed:
- Поправил тесты

KarlsonComplete committed:
- Изменил нейминг фетчера для битрикс24аккаунта.
- Добавил тест на Command для процесса завершения установки.
-Изменил контекст с debug на info в юзкейсах хендлера.

KarlsonComplete committed:
..

mesilov committed:
Merge pull request #27 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly commented on src/Bitrix24Accounts/UseCase/ChangeDomainUrl/Handler.php:
```diff
@@ -0,0 +1,42 @@
+        ]);
+
+        $accounts = $this->bitrix24AccountRepository->findByDomain($command->oldDomainUrlHost);
+        foreach ($accounts as $account) {
```
1. Спросить в "пыхтелке" почему так и норм ли такое поведение?
2. Поставить аннотацию для phpstan, чтоб заглушить ошибку

KarlsonComplete replied:
1. Спрашивал , но ничего интересного в ответ не получил, народ позадавал вопросы по коду,вроде ответил на них, но проблемы не нашли. Так что скорее всего в прошлый раз на встрече, Максим правильно предположил проблему.
2. Поставил аннотацию для phpstan ошибка ушла.

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -0,0 +1,33 @@
+            throw new InvalidArgumentException('Member ID cannot be empty.');
+        }
+
+        if (empty($this->domainUrl)) {
```
Здесь можно добавить проверку на корректность url

KarlsonComplete replied:
Добавил с помощью filter_var, или лучше регулярку ?
Для filter_var в домене должен быть протокол обязательно.

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -0,0 +1,33 @@
+
+namespace Bitrix24\Lib\Bitrix24Accounts\UseCase\InstallFinish;
+
+use Bitrix24\SDK\Core\Exceptions\InvalidArgumentException;
```
Убрать  и заменить на стандартный

KarlsonComplete replied:
Убрал

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -0,0 +1,33 @@
+        public string $applicationToken,
+        public string $memberId,
+        public string $domainUrl,
+        public ?int $bitrix24UserId,
```
@mesilov посмотри плиз сценарии, когда параметр может быть необязательным

camaxtly replied:
проверили, параметр должен быть обязательным @KarlsonComplete

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Handler.php:
```diff
@@ -0,0 +1,87 @@
+            'b24_user_id' => $command->bitrix24UserId,
+        ]);
+
+        $bitrix24Account = $this->getSingleAccountByMemberId($command->domainUrl, $command->memberId, Bitrix24AccountStatus::new, $command->bitrix24UserId);
```
1. Убрать лишние параметры.
2. Разве может пользователь быть необязательным?
3. Полагаем, что в БД порядок и все проверки уносим перед добавлением данных.

KarlsonComplete replied:
1.Убрал
3. Про какие проверки именно идет речь ?

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallStart/Handler.php:
```diff
@@ -0,0 +1,55 @@
+            'domain_url' => $command->domainUrl,
+            'member_id' => $command->memberId,
+        ]);
+
```
Добавить проверок по сценариям
1. Продублировалось событие установки от Битрикс24
2. Мы не получили событие об удалении, но приложение удалили. Потом пользователь делает повторную установку. 

Возможно, если приложение активно и пришло опять событие на установку, то сделать расширенные проверки – погенерить сценарии, пока на уровне "поговорить".

camaxtly replied:
@mesilov с тебя дока и потом по ней будем верифицировать код  сценариев/хэндлеров.

camaxtly replied:
Дока готова,  сценарии пока откладываем до реализации сущности ApplicationInstall. 
Надо начинать фиксчировать в документации @mesilov

camaxtly commented on src/Bitrix24Accounts/UseCase/RenewAuthToken/Handler.php:
```diff
@@ -0,0 +1,94 @@
+    /**
+     * @throws MultipleBitrix24AccountsFoundException
+     */
+    public function getSingleAccountByMemberId(
```
Кажется, лучше сделать свой маленький метод, чем общей-большой.

KarlsonComplete replied:
Да, в каждом handler находится свой метод, убрал лишние параметры

camaxtly commented on tests/Functional/Bitrix24Accounts/Bitrix24AccountsFetcherTest.php:
```diff
@@ -0,0 +1,102 @@
+use Symfony\Component\HttpFoundation\RequestStack;
+use Knp\Component\Pager\ArgumentAccess\RequestArgumentAccess;
+
+class Bitrix24AccountsFetcherTest extends TestCase
```
Пока удаляем до первого нужного применения

KarlsonComplete replied:
удалил

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/InstallFinish/CommandTest.php:
```diff
@@ -0,0 +1,123 @@
+
+    #[Test]
+    #[TestDox('test finish installation for Command')]
+    public function testValidCommand(): void
```
Это надо унести в unit тесты

KarlsonComplete replied:
Унес тесты в unit

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/InstallFinish/CommandTest.php:
```diff
@@ -0,0 +1,123 @@
+        $this->assertInstanceOf(Command::class, $command);
+    }
+
+    public function testEmptyApplicationToken(): void
```
в юнит

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/InstallFinish/HandlerTest.php:
```diff
@@ -0,0 +1,98 @@
+            )
+        );
+
+        $updated = $this->repository->getById($bitrix24Account->getId());
```
Забыли проверку, что статус стал "активен"

KarlsonComplete replied:
добавил

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/InstallStart/HandlerTest.php:
```diff
@@ -0,0 +1,169 @@
+     * @throws UnknownScopeCodeException
+     */
+    #[Test]
+    public function testInstallStartHappyPath(): void
```
Добавить тест на повторную установку

KarlsonComplete replied:
Добавил тест на повторную обработку, если мы пытаемся занести уже существующее сущность в базу, доктрина выбрасывает свой Exception.

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/InstallStart/HandlerTest.php:
```diff
@@ -0,0 +1,169 @@
+            $bitrix24Account->getApplicationScope(),
+            'Object not equals'
+        );
+
```
нет проверки статуса

KarlsonComplete replied:
добавил

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/Uninstall/HandlerTest.php:
```diff
@@ -0,0 +1,116 @@
+        );
+
+        $applicationToken = Uuid::v7()->toRfc4122();
+        $bitrix24Account->applicationInstalled($applicationToken);
```
Использовать билдер и создать то,что надо.

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/Uninstall/HandlerTest.php:
```diff
@@ -0,0 +1,116 @@
+
+        $this->handler->handle(new Bitrix24Accounts\UseCase\Uninstall\Command($applicationToken));
+
+        $this->expectException(Bitrix24AccountNotFoundException::class);
```
Если сработает исключение, то ты не получаешь объект $updated, а значит и не сможешь проверить его статус.

KarlsonComplete replied:
В методе getById у нас стаяла такая проверка:
```
    if (null === $account || Bitrix24AccountStatus::deleted === $account->getStatus()) {
            throw new Bitrix24AccountNotFoundException(
                sprintf('bitrix24 account not found by id %s', $uuid->toRfc4122())
            );
        }
   ```
   Мне кажется что Exception должен выкидываться только если аккаунта правда не найдено, а если же у нас аккаунт в статусе deleted , то мы его должны возвращать.

camaxtly reviewed (commented):
РЕвью 15 декабря

KarlsonComplete committed:
Поправил тесты и ошибки с phpstan

KarlsonComplete committed:
.

KarlsonComplete commented on tests/Functional/Bitrix24Accounts/UseCase/InstallStart/HandlerTest.php:
```diff
@@ -0,0 +1,169 @@
+ * @internal
+ */
+#[CoversClass(Bitrix24Accounts\UseCase\InstallStart\Handler::class)]
+class HandlerTest extends TestCase
```
Это по сути же уже не установка , а обновление , оно точно должно находится тут ?

camaxtly replied:
Мы тестируем сценарии, в рамках хэндлера делаем отдельные методы под каждый сценария

KarlsonComplete committed:
.

mesilov committed:
Merge pull request #29 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -213,7 +213,7 @@ public function applicationUninstalled(string $applicationToken): void
         }
 
-        if ($this->applicationToken !== $applicationToken) {
+        /*if ($this->applicationToken !== $applicationToken) {
```
Делаем приватные guard-ы  для каждой из проверок, что выкидывают исключения. Например guardEmpty.
Порядок – проверка пустоты, проверка соответствия токену, потому проверка на статус.

KarlsonComplete replied:
сделал, нужно ли вывести guard-ы в отдельный класс ? или оставляем их тут ?

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -32,7 +32,7 @@ public function __construct(
     {
         $account = $this->getEntityManager()->getRepository(Bitrix24Account::class)->find($uuid);
-        if (null === $account || Bitrix24AccountStatus::deleted === $account->getStatus()) {
+        if (null === $account) {
```
Нам надо, чтоб по методу getById нельзя было получить аккаунт в статусе "удален".  Желательно, это "не выбирать" в методе  find.

KarlsonComplete replied:
добавил

camaxtly commented on src/Bitrix24Accounts/Infrastructure/Doctrine/Bitrix24AccountRepository.php:
```diff
@@ -41,6 +41,16 @@ public function getById(Uuid $uuid): Bitrix24AccountInterface
         return $account;
     }
 
+    public function existsById(Uuid $uuid): bool
```
Нужна тоже проверка на статус

KarlsonComplete replied:
добавил

camaxtly commented on src/Bitrix24Accounts/UseCase/ChangeDomainUrl/Command.php:
```diff
@@ -15,5 +16,17 @@ public function __construct(
+        $this->validateDomain($newDomainUrlHost, 'newDomainUrlHost');
+    }
+
+    private function validateDomain(string $domain, string $parameterName): void
```
1. Здесь неявное отождествление domain == url. На самом деле надо разграничивать, когда домент, когда урль.
2. А что приходит от Битрикс24, что приходит, то и надо проверять. 
3. А могут быть русскоязычные домены? Эмодзи?

@mesilov посмотри плз

camaxtly replied:
От Битрикс24 приходит домен, а значит и домен мы везде использщуем. Если надо url, то "конструируем " его отдельно.

KarlsonComplete replied:
вроде написал регулярочку для проверки домена

camaxtly replied:
В соответствии с RFC 1035 и RFC 2181, длина домена URL ограничена следующими параметрами:

Общая длина полного доменного имени (FQDN, Fully Qualified Domain Name) — до 255 байтов (включая точки-разделители).
Длина одного отдельного лейбла (части домена между точками, например, "example" в "example.com") — до 63 байтов.
Это ограничение относится к доменной части URL, но сам URL (Uniform Resource Locator) может быть значительно длиннее. Например, RFC 3986 (спецификация URI) не устанавливает жесткого ограничения на общую длину URL, но на практике браузеры и серверы часто ограничивают URL до 2048–8192 символов.

camaxtly replied:
У Битрикс24 можно использовать кастомный домен –  а значит может быть больше 3-х уровней.  И уточнить, сколько символов может быть в домене первого уровня.

Опциональ вынесли домен в Value Object

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -25,9 +25,12 @@ private function validate(): void
-
-        if (empty($this->domainUrl)) {
-            throw new InvalidArgumentException('Domain URL cannot be empty.');
+        /*$pattern = '/^(https?:\/\/)?([a-z0-9-]+\.[a-z]{2,})(\/[^\s]*)?$/i';
```
Удаляем отладку

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -25,9 +25,12 @@ private function validate(): void
+        if (!preg_match($pattern, $this->domainUrl)) {
+            throw new InvalidArgumentException('Domain URL is not valid.');
+        }*/
+        if (!filter_var($this->domainUrl, FILTER_VALIDATE_URL)) {
```
@mesilov посмотри как правильно назваются части URL, и что по факту к нам приходит? url/domain/host

camaxtly replied:
Приходит domain @KarlsonComplete

KarlsonComplete replied:
переписал добавил регулярочку

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallStart/Command.php:
```diff
@@ -7,17 +7,48 @@
+            throw new InvalidArgumentException('Member ID must be a non-empty string.');
+        }
+
+        if (!filter_var($this->domainUrl, FILTER_VALIDATE_URL)) {
```
Поправить, как только выровняем представление по url

camaxtly replied:
Согласовали, у нас domain, переписать проверку

KarlsonComplete replied:
поправил

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallStart/Command.php:
```diff
@@ -7,17 +7,48 @@
+            throw new InvalidArgumentException('Application version must be a positive integer.');
+        }
+
+        if (!is_string($this->authToken->accessToken)) {
```
Не надо проверять токен, так как он или уже пришле валидный или должны "упать" раньше

KarlsonComplete replied:
убрал проверку

camaxtly commented on tests/Functional/Bitrix24Accounts/Builders/Bitrix24AccountBuilder.php:
```diff
@@ -53,7 +53,7 @@ public function __construct()
         $this->isBitrix24UserAdmin = true;
         $this->memberId = Uuid::v4()->toRfc4122();
-        $this->domainUrl = Uuid::v7()->toRfc4122() . '-test.bitrix24.com';
+        $this->domainUrl = 'https://'.Uuid::v7()->toRfc4122() . '-test.bitrix24.com';
```
Не забыть поправить и тут тоже

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/InstallStart/HandlerTest.php:
```diff
@@ -156,7 +156,8 @@ public function testInstallStartHappyPath(): void
         );
 
-       // var_dump($this->eventDispatcher->getOrphanedEvents());
+        $this->assertEquals('new',$bitrix24Account->getStatus()->value);
```
Bitrix24AccountStatus использовать для получения литерала, а не просто 'new'.
Сравниваем enum, а не строки как сейчас.

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/InstallStart/HandlerTest.php:
```diff
@@ -166,4 +167,44 @@ public function testInstallStartHappyPath(): void
     }
+
+    #[Test]
+    public function testReinstallApplication(): void
```
Переустановка идет по 2-м сценариям
1 – удалили приложение и мы обработали удаление, в базе есть аккаунт, но он в статусе "удален".
2 - мы пропустили события удаления, для нас приложение активно.

@mesilov а какие токены приезжают при переустановке приложения? АпликейшенТокен один и тот же? Установка / удаление / переустановка 1 админом / переустановка разными админами

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/Uninstall/HandlerTest.php:
```diff
@@ -56,28 +57,47 @@ class HandlerTest extends TestCase
+        $applicationToken = Uuid::v7()->toRfc4122();
+        $id = Uuid::v7();
+        $memberId = Uuid::v7()->toRfc4122();
         $bitrix24Account = new Bitrix24Account(
```
Добавляем в билдер метод withApplicationToken  и используем его.

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/InstallStart/CommandTest.php:
```diff
@@ -0,0 +1,136 @@
+
+    private Flusher $flusher;
+
+    private Bitrix24AccountRepositoryInterface $repository;
```
Удалить репезиторий и все, что с ним связано.

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/InstallFinish/CommandTest.php:
```diff
@@ -2,24 +2,24 @@
 declare(strict_types=1);
 
-namespace Bitrix24\Lib\Tests\Functional\Bitrix24Accounts\UseCase\InstallFinish;
+namespace Bitrix24\Lib\Tests\Unit\Bitrix24Accounts\UseCase\InstallFinish;
```
В юнит тестах могут быть только InMemory репозитории, если тебе надо сохранять в БД, то тест уже функциональный. 
Для тестирования команды, функциональный тест не нужен – тебе надо только проверить, что в правильных ключах команды корректные данные.

camaxtly reviewed (commented):
23-12-2024

KarlsonComplete committed:
Правки

KarlsonComplete committed:
Правки

KarlsonComplete committed:
Добавил 2 юнит теста

KarlsonComplete committed:
Добавил 2 юнит теста

KarlsonComplete committed:
Правки в тест на повторную установку аккаунта
Удалил тест на обновление токена так как там нечего проверять

KarlsonComplete committed:
Правки в тест на повторную установку аккаунта

camaxtly committed:
Merge pull request #32 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -211,24 +215,51 @@ public function applicationUninstalled(string $applicationToken): void
+    }
 
-        $this->accountStatus = Bitrix24AccountStatus::deleted;
+    private function guardStatusIsActive()
```
Нейминг – ApplicationIsActive

camaxtly commented on src/Bitrix24Accounts/Entity/Bitrix24Account.php:
```diff
@@ -117,89 +117,93 @@ public function renewAuthToken(RenewedAuthToken $renewedAuthToken): void
-     */
-    #[Override]
-    public function applicationUninstalled(string $applicationToken): void
+    private function guardEmpty($applicationToken)
```
Нейминг – EmptyToken

camaxtly commented on src/Bitrix24Accounts/UseCase/RenewAuthToken/Command.php:
```diff
@@ -0,0 +1,15 @@
+{
+    public function __construct(
+        public RenewedAuthToken $renewedAuthToken,
+        public ?int $bitrix24UserId = null,
```
По логике это обязательный параметр

camaxtly commented on src/Bitrix24Accounts/UseCase/Uninstall/Command.php:
```diff
@@ -0,0 +1,27 @@
+
+    private function validate(): void
+    {
+        if (empty($this->applicationToken) || !Uuid::isValid($this->applicationToken)) {
```
Убрать проверку на uuid, токен это не uuid

camaxtly commented on tests/Functional/Bitrix24Accounts/UseCase/InstallStart/HandlerTest.php:
```diff
@@ -0,0 +1,213 @@
+     * @throws UnknownScopeCodeException
+     */
+    #[Test]
+    public function testInstallStartHappyPath(): void
```
Методы именуем по сценариям - полностью новая/чистая установка, переустановка и т/д

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/InstallFinish/CommandTest.php:
```diff
@@ -0,0 +1,95 @@
+        $this->assertInstanceOf(Command::class, $command);
+    }
+
+    public function testEmptyApplicationToken(): void
```
Мне кажется, тут отлично подойдет дата провайдер

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/Uninstall/CommandTest.php:
```diff
@@ -0,0 +1,24 @@
+    {
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Empty application token or invalid application token.');
```
"or invalid application token" –  мы не знаем, что такое невалидный токен, можем проверить только на пустоту, так как это строка.

camaxtly reviewed (commented):
6-01-2025

KarlsonComplete committed:
- Добавил датаправайдер в одном из тестов на команду.
- Внес еще некоторые мелкие правки

KarlsonComplete committed:
- Добавил генератор для всех тестов

KarlsonComplete committed:
- юзанул cs-fixer

KarlsonComplete committed:
- юзанул cs-fixer для тестов

camaxtly committed:
Merge pull request #34 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly commented on src/Bitrix24Accounts/ReadModel/Bitrix24AccountFetcher.php:
```diff
@@ -0,0 +1,37 @@
+use Knp\Component\Pager\Pagination\PaginationInterface;
+use Knp\Component\Pager\PaginatorInterface;
+
+class Bitrix24AccountFetcher
```
договорились пока удалить

KarlsonComplete replied:
удалил

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/Uninstall/CommandTest.php:
```diff
@@ -0,0 +1,45 @@
+
+    public static function dataForCommand(): \Generator
+    {
+        yield 'validApplicationToken' => [
```
Переименовать emptyAplicationToken и сделать тест для валидного токена

KarlsonComplete replied:
добавил

KarlsonComplete committed:
- Правки для домена
- Преобразовал тест для домена

KarlsonComplete committed:
- Правки для фукнциональных тестов
- Добавил тест для проверки функции

KarlsonComplete committed:
- Пытаюсь править для функциональных тестов в github actions

KarlsonComplete committed:
- Тест на проверку функции домена

KarlsonComplete committed:
Update tests-functional.yml

KarlsonComplete committed:
- Правим github actions

KarlsonComplete committed:
- Правим github actions

KarlsonComplete committed:
- Правим github actions

KarlsonComplete committed:
- Юзанули ректор

KarlsonComplete committed:
- Правки для github actions

KarlsonComplete committed:
- Правки для github actions

KarlsonComplete committed:
- Правки для github actions 2

KarlsonComplete committed:
- Правки для github actions 3

KarlsonComplete committed:
- Правки для тестов
- Правки для сs fixer

KarlsonComplete committed:
- Правки для сs fixer

camaxtly committed:
Merge pull request #36 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/ChangeDomainUrl/CommandTest.php:
```diff
@@ -0,0 +1,61 @@
+
+    public static function dataForCommand(): \Generator
+    {
+        $invalidOldDomain = 'invalid_domain.com';
```
Переписать тест-провайдер, у тебя просто перечислены переменные, а ожидается что-то вроде массива и цикла по нему с yeld.

KarlsonComplete replied:
сделано

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -0,0 +1,52 @@
+        }
+    }
+
+    private function validateDomain(string $domain): void
```
Выносим домен в VO - принятое решение

KarlsonComplete replied:
сделано

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -0,0 +1,52 @@
+
+    private function validate(): void
+    {
+        if ('' === $this->applicationToken || '0' === $this->applicationToken) {
```
Если на вход строка, то мы и работаем как со строкой. Числа тут не нужны

KarlsonComplete replied:
сделано

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallStart/Command.php:
```diff
@@ -0,0 +1,64 @@
+
+    private function validate(): void
+    {
+        if (empty($this->uuid)) {
```
Скорее всего тут не нужна проверка

KarlsonComplete replied:
сделано

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallStart/Command.php:
```diff
@@ -0,0 +1,64 @@
+            throw new \InvalidArgumentException('Bitrix24 User ID must be a positive integer.');
+        }
+
+        if (!is_string($this->memberId) || ('' === $this->memberId || '0' === $this->memberId)) {
```
Убрать ненужные проверки

KarlsonComplete replied:
сделано

camaxtly commented on tests/Functional/Bitrix24Accounts/Builders/Bitrix24AccountBuilder.php:
```diff
@@ -0,0 +1,123 @@
+        return $this;
+    }
+
+    public function build(): AggregateRootEventsEmitterInterface&Bitrix24AccountInterface
```
& – зачем тут надо?  AggregateRootEventsEmitterInterface вообще лишнее.

По идее тут должен быть Bitrix24Account

KarlsonComplete replied:
поменял

camaxtly commented on tests/Unit/Bitrix24Accounts/DomainCheckerTest.php:
```diff
@@ -0,0 +1,83 @@
+
+    }
+
+    public static function dataForCommand(): \Generator
```
Разнести на 2 провайдера и 2 теста

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/Uninstall/CommandTest.php:
```diff
@@ -0,0 +1,59 @@
+        }
+    }
+
+    public static function dataForCommand(): \Generator
```
Опционально разнести на позитивный и негативный сценарии

KarlsonComplete replied:
сделано

camaxtly reviewed (commented):
02-02-2025

KarlsonComplete committed:
- Перенес функцию на проверку домена в отдельный тест
- В тестах поправил создание команды

KarlsonComplete committed:
- Поправил тесты

KarlsonComplete committed:
- Поправил сущность аккаунта , добавил параметры для метода updateApplicationVersion
- Юзанул cs-fixer
- Юзанул ректор
- Добавлена сущность Domain

camaxtly committed:
Merge pull request #37 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly commented on src/Bitrix24Accounts/UseCase/ChangeDomainUrl/Command.php:
```diff
@@ -0,0 +1,22 @@
+        Domain $oldDomain,
+        Domain $newDomain
+    ) {
+        $this->oldDomain = $oldDomain->getValue();
```
Используем объекты

camaxtly commented on src/Bitrix24Accounts/ValueObjects/Domain.php:
```diff
@@ -0,0 +1,39 @@
+
+readonly class Domain
+{
+    private string $value;
```
А может быть public?

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/ChangeDomainUrl/CommandTest.php:
```diff
@@ -0,0 +1,116 @@
+    ): void {
+
+        $exceptionCount = 0;
+        foreach ($arrDomains as $arrDomain) {
```
Уходим от цикла внутри 1 текста и переходим к вызову нескольких тестов и провайдер данных

camaxtly commented on tests/Unit/Bitrix24Accounts/UseCase/Uninstall/CommandTest.php:
```diff
@@ -0,0 +1,62 @@
+        string $applicationToken,
+    ): void {
+        $command = new Command($applicationToken);
+        $this->assertInstanceOf(Command::class, $command);
```
Изменить проверку что токен из команды равен переданному токену

camaxtly commented on src/Bitrix24Accounts/UseCase/InstallFinish/Command.php:
```diff
@@ -0,0 +1,37 @@
+            throw new \InvalidArgumentException('Application token cannot be empty.');
+        }
+
+        if ('' === $this->memberId || '0' === $this->memberId) {
```
Выпиливаем все проверки на строковые 0

KarlsonComplete commented on src/Bitrix24Accounts/UseCase/Uninstall/Command.php:
```diff
@@ -0,0 +1,24 @@
+
+    private function validate(): void
+    {
+        if ('' === $this->applicationToken || '0' === $this->applicationToken) {
```
сделано

KarlsonComplete committed:
- Поправил домен
- Пофиксил юнит тесты на проверки и на домен

KarlsonComplete committed:
- Поправил домен

KarlsonComplete committed:
- Добавил useCase на обновления версии и написал тесты

camaxtly committed:
Merge pull request #38 from KarlsonComplete/add-docker-structure

Add docker structure

camaxtly reviewed (approved):
16-02-2025
